src/caches/abstractbeliefpropagationcache.jl

```diff
@@ -241,11 +286,11 @@ function update(
   edge_groups::Vector{<:Vector{<:PartitionEdge}};
   kwargs...,
 )
-  new_mts = copy(messages(bpc))
+  new_mts = Dictionary()
```

Comment:
```suggestion
  new_mts = empty(messages(bpc))
```
That would help to preserve the type of the message cache.